android/src/main/java/com/example/flutter_file_downloader/MethodCallHandlerImpl.java

```diff
@@ -157,18 +157,19 @@ public void onResult(StoragePermission permission) {
     }
 
     private void onStartDownloadingFile(StoreHelper helper) {
-        try {
-            if (!permissionManager.hasPermission(context)) {
-                onRequestPermission(helper, false);
-                return;
-            }
-        } catch (PermissionUndefinedException e) {
-            helper.result.error(
-                    ErrorCodes.permissionDefinitionsNotFound.toString(),
-                    ErrorCodes.permissionDefinitionsNotFound.toDescription(),
-                    null);
-            return;
-        }
+        // Permission will handle by other packages (examples:https://pub.dev/packages/permission_handler)
+        //        try {
+        //            if (!permissionManager.hasPermission(context)) {
+        //                onRequestPermission(helper, false);
+        //                return;
+        //            }
+        //        } catch (PermissionUndefinedException e) {
+        //            helper.result.error(
+        //                    ErrorCodes.permissionDefinitionsNotFound.toString(),
+        //                    ErrorCodes.permissionDefinitionsNotFound.toDescription(),
+        //                    null);
+        //            return;
+        //        }
```

Comment:
private void onStartDownloadingFile(StoreHelper helper, { boolean autoHandlePermission = true }) {
        if (autoHandlePermission) {
        try {
            if (!permissionManager.hasPermission(context)) {
                onRequestPermission(helper, false);
                return;
            }
        } catch (PermissionUndefinedException e) {
            helper.result.error(
                    ErrorCodes.permissionDefinitionsNotFound.toString(),
                    ErrorCodes.permissionDefinitionsNotFound.toDescription(),
                    null);
            return;
        }
        }
  }.

Reply:
May I ask how exactly is this helpful?
What if the user set it to no?
The app might crash

Reply:
May I ask how exactly is this helpful?
What if the user set it to no?
The app might crash

Reply:
Why:
flutter_file_downloader use the logic permission handler & don’t work with Android 13 => he want to delegate for [permission_handler](https://pub.dev/packages/permission_handler) package do it instead.

Reply:
I updated code @abdallah-odeh , This is helpful for someone need to handle permission by themself.
If user set autoHandlePermission = false & don't handle permission by themself, will have the logs & not crash app 🙏

Reply:
Can y help me to reviews to @abdallah-odeh 👍